cpp/src/io/parquet/parquet.hpp

```diff
@@ -45,6 +45,104 @@ struct file_ender_s {
   uint32_t magic;
 };
 
+// thrift generated code simplified.
+struct StringType {
+};
+struct MapType {
+};
+struct ListType {
+};
+struct EnumType {
+};
+struct DecimalType {
+  int32_t scale     = 0;
+  int32_t precision = 0;
+};
+struct DateType {
+};
+
+struct MilliSeconds {
+};
+struct MicroSeconds {
+};
+typedef struct _TimeUnit__isset {
+  _TimeUnit__isset() : MILLIS(false), MICROS(false) {}
+  bool MILLIS;
+  bool MICROS;
+} _TimeUnit__isset;
+
+struct TimeUnit {
+  _TimeUnit__isset __isset;
+  MilliSeconds MILLIS;
+  MicroSeconds MICROS;
+};
+
+struct TimeType {
+  bool isAdjustedToUTC = false;
+  TimeUnit unit;
+};
+struct TimestampType {
+  bool isAdjustedToUTC = false;
+  TimeUnit unit;
+};
+struct IntType {
+  int8_t bitWidth = 0;
+  bool isSigned   = false;
+};
+struct NullType {
+};
+struct JsonType {
+};
+struct BsonType {
+};
+
+// thrift generated code simplified.
+typedef struct _LogicalType__isset {
+  _LogicalType__isset()
+    : STRING(false),
+      MAP(false),
+      LIST(false),
+      ENUM(false),
+      DECIMAL(false),
+      DATE(false),
+      TIME(false),
+      TIMESTAMP(false),
+      INTEGER(false),
+      UNKNOWN(false),
+      JSON(false),
+      BSON(false)
+  {
+  }
+  bool STRING;
+  bool MAP;
+  bool LIST;
+  bool ENUM;
+  bool DECIMAL;
+  bool DATE;
+  bool TIME;
+  bool TIMESTAMP;
+  bool INTEGER;
+  bool UNKNOWN;
+  bool JSON;
+  bool BSON;
+} _LogicalType__isset;
+
+struct LogicalType {
+  _LogicalType__isset __isset;
```

Comment:
As far as I know identifiers which start with an underscore are reserved for the implementation. I have not seen it be a problem because yet but its always good to be careful.

From gnu libc manual found [here](https://www.gnu.org/savannah-checkouts/gnu/libc/manual/html_node/Reserved-Names.html):
> In addition to the names documented in this manual, reserved names include all external identifiers (global functions and variables) that begin with an underscore (‘_’) and all identifiers regardless of use that begin with either two underscores or an underscore followed by a capital letter are reserved names. This is so that the library and header files can define functions, variables, and macros for internal purposes without risk of conflict with names in user programs.